src/app-manager.ts

```diff
@@ -64,26 +72,21 @@ export class AppManager {
   /**************************/
 
   /** Add an api to the AppModule */
-  private activateApi(json: ModuleProfile, api: API) {
+  private activateApi<M extends Api>(json: ModuleProfile<M>, api: API<M>) {
     const events = json.events || []
     events.forEach(event => {
-      if (event in api) {
-        api[event].on((value: any) => {
-          this.broadcast(api.type, event, value)
-        })
-      }
+      if (!api.events) return
+      api.events.on(event, (value: any) => this.broadcast(api.type, event, value))

```

Comment:
does we need the template here on the call to `broadcast`

Reply:
What do you mean by "template" ?